src/javascript/app/pages/user/account/settings/personal_details.js

```diff
@@ -54,6 +59,70 @@ const PersonalDetails = (() => {
         $('#missing_details_notice').setVisibility(!!has_missing_field);
     };
 
+    const populateChangeableFields = () => {
+        if (!has_changeable_fields) return;
+
+        const landing_companies = State.getResponse('landing_company');
+        const changeable        = landing_companies.financial_company.changeable_fields;
+        if (changeable && changeable.only_before_auth) {
+            changeable_fields = changeable.only_before_auth;
+        }
+    };
+
+    /**
+     * Remove labels and static fields and replace them with input when fields are changeable.
+     *
+     * @param {get_settings} to prepopulate some of the values.
+     */
+    const displayChangeableFields = (get_settings) => {
+        if (!has_changeable_fields) return;
+        changeable_fields.forEach(field => {
+            CommonFunctions.getElementById(`row_${field}`).setVisibility(1);
+        });
+
+        if (['first_name', 'last_name', 'salutation'].some(key => changeable_fields.includes(key))) {
+            CommonFunctions.getElementById('row_name').setVisibility(0);
+            CommonFunctions.getElementById('first_name').setVisibility(1);
+            CommonFunctions.getElementById('last_name').setVisibility(1);
+            CommonFunctions.getElementById('salutation').setVisibility(1);
```

Comment:
I meant if we loop the array, we can combine these three lines to `CommonFunctions.getElementById(looping_item).setVisibility(1);`, and handle setting row_name to invisible once

Reply:
but if you think this is better then keep it